src/pdf/ShoppingList/ShoppingListPDF.tsx

```diff
@@ -0,0 +1,228 @@
+            </View>
+            <View style={styles.tableQuantity}>
+                <Text style={styles.normalText}>{item.quantity}</Text>
+            </View>
+            <View style={styles.tableNotes}>
+                <Text style={styles.normalText}>{item.notes}</Text>
+            </View>
+            <View style={styles.tableDone}></View>
+        </View>
+    );
+};
+
+const DisplayItemsList: React.FC<{ itemsList: Item[] }> = ({ itemsList }) => {
+    return (
+        <View>
+            {itemsList.map((item, index) => (
+                <ItemToRow {...item} key={index} />
+            ))}
+        </View>
+    );
+};
+
+const formatCamelCaseKey = (objectKey: string): string => {
+    return objectKey.replaceAll(/([a-z])([A-Z])/g, "$1 $2");
+};
+
+const DisplayAsBlock: React.FC<BlockProps> = (data: BlockProps) => {
+    return (
+        <View style={styles.infoCell}>
+            {Object.keys(data).map((propKey, index) => (
+                <OneLine key={index} header={formatCamelCaseKey(propKey)} value={data[propKey]} />
+            ))}
+        </View>
+    );
+};
+
+const FormatClientCell: React.FC<{ propKey: string; propValue: string }> = ({
+    propKey,
+    propValue,
+}) => {
+    return (
+        <View style={styles.infoCell}>
+            <OneLine header={formatCamelCaseKey(propKey)} value={propValue} />
+        </View>
+    );
+};
+
+const DisplayClientSummary: React.FC<ClientSummary> = (clientSummary) => {
+    return (
+        <>
+            <View style={[styles.flexRow, styles.infoBlock]}>
+                <FormatClientCell propKey="name" propValue={clientSummary.name} />
+                <FormatClientCell propKey="contact" propValue={clientSummary.contact} />
+            </View>
+            <View style={[styles.flexRow, styles.infoBlock]}>
+                <FormatClientCell propKey="address" propValue={clientSummary.address} />
+                <FormatClientCell
+                    propKey="extraInformation"
+                    propValue={clientSummary.extraInformation}
+                />
+            </View>
+        </>
+    );
+};
+
+const ShoppingListPDF: React.FC<ShoppingListPDFProps> = ({
+    postcode,
+    parcelInfo,
+    clientSummary,
+    householdSummary,
+    requirementSummary,
+    itemsList,
+    endNotes,
+}) => {
+    return (
+        <Document>
+            <Page size="A4">
+                <View style={styles.paper}>
+                    <View style={{ marginTop: "0.75in" }} fixed />
+                    <View style={[styles.flexRow, styles.pdfHeader]}>
+                        <View style={styles.flexColumn}>
+                            <Text style={styles.title}>Shopping List</Text>
+                            <Text style={styles.subtitle}>POSTCODE: {postcode}</Text>
+                        </View>
+                        {/* eslint-disable-next-line jsx-a11y/alt-text -- React-PDF doesn't Image have alt text property*/}
```

Comment:
super minor but is it a typo? "React-PDF Image doesn't have ..."

Reply:
nice catch, thanks :)